### PR TITLE
Flake8

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
**Description**

Fix part of the Flake8 issues, usually minor ones.

This will help address more serious alerts such as:
```
./gammapy/data/tests/test_observations.py:226:1: F811 redefinition of unused 'test_observation' from line 25
```

Partially addresses #3551.

**Dear reviewer**

Ready for review.

### CODE REVIEW

**Comment by adonath** [comment]
@DimitriPapadopoulos Just to clarify: I haven't forgot about this PR, but I'd prefer that you re-run the tools shortly before the next release, when most other PRs are merged.. So I'll come back to you in the week before Nov. 19th...does this work for you?

File: setup.cfg [hunk 1]
```diff
@@ -109,3 +109,10 @@ line_length=88
 skip = ./.git
 ignore-words = dev/codespell/ignore-words.txt
 exclude-file = dev/codespell/exclude-file.txt
+
+[flake8]
+# https://flake8.pycqa.org/en/latest/user/error-codes.html
+# W503: line break before binary operator
+ignore = W503
+exclude = extern
+>>>>>>> Fix some Flake8 alerts
```
Comment (adonath) [inline setup.cfg:118]: Here is a left over from a merge conflict...please remove.

Comment (DimitriPapadopoulos) [reply setup.cfg]: Done, but I need to review the results of the merge one more time.

**Comment by DimitriPapadopoulos** [comment]
What does **`#%%`** mean in the following comments?
https://github.com/gammapy/gammapy/blob/ecf3949e4a7ffbf97881985c78307fde1d0acd5d/examples/models/spatial/plot_constant.py#L10-L13
https://github.com/gammapy/gammapy/blob/ecf3949e4a7ffbf97881985c78307fde1d0acd5d/examples/models/spatial/plot_constant.py#L27-L30

**Comment by DimitriPapadopoulos** [comment]
Other than the above remark about `# %%`, I have reviewed the commits once more and they look good to me.

**Comment by registerrier** [comment]
> What does **`# %%`** mean in the following comments?
> 
> https://github.com/gammapy/gammapy/blob/ecf3949e4a7ffbf97881985c78307fde1d0acd5d/examples/models/spatial/plot_constant.py#L10-L13
> 
> 
> https://github.com/gammapy/gammapy/blob/ecf3949e4a7ffbf97881985c78307fde1d0acd5d/examples/models/spatial/plot_constant.py#L27-L30

The `# %%` is a directive to include rST in the sphinx-gallery example. See:
https://sphinx-gallery.github.io/stable/syntax.html#embed-rst-in-your-example-python-files

**Comment by DimitriPapadopoulos** [comment]
Thank you. So it was correct to change the second `#%%` into `# %%`.

File: docs/irf/plot_fermi_psf.py [hunk 2]
```diff
@@ -11,4 +11,4 @@
 # .to_image() computes the exposure weighted mean PSF
 kernel = psf.get_psf_kernel(geom=geom).to_image()
 
-kernel.psf_kernel_map.plot();
\ No newline at end of file
+kernel.psf_kernel_map.plot()
```
Comment (adonath) [inline docs/irf/plot_fermi_psf.py:14]: Taking it strictly I think this is a false positive: the semicolon suppresses the output returned by the `.plot()` method. But I think we are nowhere uniform in Gammapy about this, so I'm fully OK to keep this for now.

**Review by adonath — APPROVED** [review]
Thanks @DimitriPapadopoulos, I'll go ahead and merge now.

**Pull request merged**